acir/src/circuit/opcodes.rs

```diff
@@ -188,15 +188,23 @@ impl std::fmt::Display for Opcode {
                     bits.last().unwrap().witness_index(),
                 )
             }
-            Opcode::Directive(Directive::Log(info)) => match info {
-                LogInfo::FinalizedOutput(output_string) => write!(f, "Log: {output_string}"),
-                LogInfo::WitnessOutput(witnesses) => write!(
-                    f,
-                    "Log: _{}..._{}",
-                    witnesses.first().unwrap().witness_index(),
-                    witnesses.last().unwrap().witness_index()
-                ),
-            },
+            Opcode::Directive(Directive::Log {
+                is_trace,
+                output_info,
+            }) => {
+                let is_trace_display = if *is_trace { "trace" } else { "println" };
+                match output_info {
+                    LogOutputInfo::FinalizedOutput(output_string) => {
+                        write!(f, "Log: {output_string}, log type {is_trace_display}")
```

Comment:
Appending `, log type {}` after the log rather than before seems odd to me. I'm not aware of a logging framework that does not format logs with the log type before the log itself: `<log type>: <log>`. It also seems odd to me that a `println` would actually print "println" before every message. Am I reading this wrong? Ideally if we do want to print the log types I'd like logs to be displayed as `trace: my trace string` and `my println string` omitting the println entirely.

Reply:
I think you might be misreading this, this isn't quite the log output itself but the display representation of `Log` opcode. We should then display information on how we're logging the data rather than just printing out the message which will be logged at execution time.